x/attribute/client/cli/cli_test.go

```diff
@@ -1197,3 +1237,129 @@ func (s *IntegrationTestSuite) TestPaginationWithPageKey() {
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			expectErr:    `invalid address: must be either an account address or scope metadata address: "not-a-address"`,
+			respType:     &sdk.TxResponse{},
+			expectedCode: 0,
+		},
+		{
+			name: "update expire date, should fail incorrect date",
+			cmd:  cli.NewUpdateAccountAttributeExpirationCmd(),
+			args: []string{
+				"expiration.attribute",
+				s.account2Addr.String(),
+				"string",
+				"test value",
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.testnet.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			expectErr:    `unable to parse time (0001-01-01 00:00:00 +0000 UTC) required format is RFC3339 (2006-01-02T15:04:05Z07:00) , parsing time "test value" as "2006-01-02T15:04:05Z07:00": cannot parse "test value" as "2006"`,
+			respType:     &sdk.TxResponse{},
+			expectedCode: 0,
+		},
+		{
+			name: "update expire date, should succeed",
+			cmd:  cli.NewUpdateAccountAttributeExpirationCmd(),
+			args: []string{
+				"expiration.attribute",
+				s.account2Addr.String(),
+				"test value",
+				"2050-01-15T00:00:00Z",
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.testnet.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			respType:     &sdk.TxResponse{},
+			expectedCode: 0,
+		},
+		{
+			name: "update expire date, should succeed removes expiration",
+			cmd:  cli.NewUpdateAccountAttributeExpirationCmd(),
+			args: []string{
+				"expiration.attribute",
+				s.account2Addr.String(),
+				"test value",
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.testnet.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			respType:     &sdk.TxResponse{},
+			expectedCode: 0,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			clientCtx := s.testnet.Validators[0].ClientCtx
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, tc.cmd, tc.args)
+
+			if len(tc.expectErr) > 0 {
+				s.Require().EqualError(err, tc.expectErr)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code)
```

Comment:
I notice this pattern all over the place in tests this is because we do a lot of copy and paste for the boiler plate code from older tests.  I wonder if we should make an issue to get these cleaned?